addons/project/static/src/views/burndown_chart/burndown_chart_search_model.js

```diff
@@ -42,9 +45,11 @@ export class BurndownChartSearchModel extends SearchModel {
      * @override
      */
     deactivateGroup(groupId) {
-        // Prevent removing Date & Stage group by from the search
-        if (this.searchItems[this.stageIdSearchItemId].groupId == groupId && this.searchItems[this.dateSearchItemId].groupId) {
-            this._addGroupByNotification(_t("Date and Stage"));
+        // Prevent removing 'Date & Stage' and 'Date & is closed' group by from the search
+        if (this.searchItems[this.dateSearchItemId].groupId == groupId) {
+            if (this.query.some(queryElem => queryElem.searchItemId === this.stageIdSearchItemId) || this.query.some(queryElem => queryElem.searchItemId === this.isClosedSearchItemId)){
```

Comment:
I am wondering if it is not equivalent to:
```suggestion
            if (this.query.some(queryElem => [this.stageIdSearchItemId, 
 this.isClosedSearchItemId].includes(queryElem.searchItemId)) {
```

What do you think? 🤔

Reply:
It seems to be working. I don't have enough knowledge on JS to know if it's good or bad practice. But i do like it better than the old code because it's shorter and more readable, so i've made the changes.

Reply:
`some` function for array in JS is like `any` function in python.
and so instead of doing 2 times the loop, once to check if `this.stageIdSearchItemId` is in the array and another to check if `this.isClosedSearchItemId` we could check the both in single loop since your operator was `OR`. And so it equivalent to say "search in `this.query` array and if you find `this.stageIdSearchItemId` or `this.isClosedSearchItemId` then return true".

In python we could write my suggestion like this:
```py
if any(query_elem in [stage_id_search_item_id, is_closed_search_item_id] for query_item in query)
```
where
- `query` is an array (correspond to `this.query` in JS)
- `stage_id_search_item_id` contains the id of the Stage_id filter/group by in the search view (correspond to `this.stageIdSearchItemId` in the JS code)
- `is_closed_search_item_id` contains the id of of the is_closed filter/group by in the search view (correspond to `this.isClosedSearchItemId`)